XiEditor/EditViewController.swift

```diff
@@ -109,8 +109,13 @@ class EditViewController: NSViewController, EditViewDataSource, FindDelegate, Sc
     // visible scroll region
     var visibleLines: LineRange = 0..<0
 
-    // TODO: should be an option in the user preferences
-    var scrollPastEnd = false
+    var scrollPastEnd = false {
+        didSet {
+            if scrollPastEnd != oldValue {
+                redrawEverything()
```

Comment:
I think this is fine, but technically it probably only requires `updateEditViewHeight`